### PR TITLE
feat(tracing): support http server error status configuration

### PR DESCRIPTION
**Pull request by mabdinur**
DD_TRACE_HTTP_SERVER_ERROR_STATUSES can only be enabled via the programmatic api. This PR adds an environment variable that can be used to update this configuration.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/tracer/test_trace_utils.py [hunk 1]
```diff
@@ -505,6 +505,24 @@ def test_set_http_meta_custom_errors(mock_log, span, int_config, error_codes, st
...
+    config.http_server.error_statuses = "404-412"
+
+    config._add("myint", dict())
+    with tracer.trace("error") as span1:
+        set_http_meta(span1, config.myint, status_code=405)
+        assert span1.error == 1
+
+    with tracer.trace("noterror") as span2:
+        set_http_meta(span2, config.myint, status_code=403)
+        assert span2.error == 0
```
Comment (mabdinur) [inline tests/tracer/test_trace_utils.py:523]:
```suggestion
        assert span2.error == 0

    with tracer.trace("noterror") as span2:
        set_http_meta(span2, config.myint, status_code=403)
        assert span2.error == 0
```

File: docs/configuration.rst [hunk 2]
```diff
@@ -461,6 +461,14 @@ The following environment variables for the tracer are supported:
...
      default: True
      description: Send query strings in http.url tag in http server integrations.
+  
+
+  DD_TRACE_HTTP_CLIENT_ERROR_STATUSES
+    type: String
+    default: "500-599"
+    description: |
+        Comma-separated list of HTTP status codes that should be considered errors when returned by an HTTP client request.
+        The status codes are used to set the ``error`` field on the span.
```
Comment (mabdinur) [inline docs/configuration.rst:471]:
```suggestion
   DD_TRACE_HTTP_CLIENT_ERROR_STATUSES
     type: String
     default: "500-599"
     description: |
        Comma-separated list of HTTP status codes that should be considered errors when returned by an HTTP client request.
        The status codes are used to set the ``error`` field on the span.
```

File: docs/configuration.rst [hunk 3]
```diff
@@ -461,6 +461,14 @@ The following environment variables for the tracer are supported:
      type: Boolean
      default: True
      description: Send query strings in http.url tag in http server integrations.
+
```
Comment (mabdinur) [inline docs/configuration.rst:464]:
```suggestion
```

File: tests/tracer/test_trace_utils.py [hunk 4]
```diff
@@ -505,6 +505,28 @@ def test_set_http_meta_custom_errors(mock_log, span, int_config, error_codes, st
         mock_log.exception.assert_not_called()
 
 
+@pytest.mark.subprocess(env={"DD_TRACE_HTTP_CLIENT_ERROR_STATUSES": "404-412"})
```
Comment (mabdinur) [inline tests/tracer/test_trace_utils.py:508]:
```suggestion
@pytest.mark.subprocess(env={"DD_TRACE_HTTP_SERVER_ERROR_STATUSES": "404-412"})
```

File: docs/configuration.rst [hunk 5]
```diff
@@ -462,6 +462,13 @@ The following environment variables for the tracer are supported:
      default: True
      description: Send query strings in http.url tag in http server integrations.
 
+   DD_TRACE_HTTP_CLIENT_ERROR_STATUSES
```
Comment (mabdinur) [inline docs/configuration.rst:465]:
```suggestion
   DD_TRACE_HTTP_SERVER_ERROR_STATUSES
```

File: ddtrace/settings/config.py [hunk 6]
```diff
@@ -326,7 +326,7 @@ class Config(object):
     """
 
     class _HTTPServerConfig(object):
-        _error_statuses = "500-599"  # type: str
+        _error_statuses = os.getenv("DD_TRACE_HTTP_SERVER_ERROR_STATUSES", "500-599")  # type: str
```
Comment (mabdinur) [inline ddtrace/settings/config.py:329]:
```suggestion
        _error_statuses = os.getenv("DD_TRACE_HTTP_SERVER_ERROR_STATUSES", "500-599")  # type: str
```

File: docs/configuration.rst [hunk 7]
```diff
@@ -462,6 +462,13 @@ The following environment variables for the tracer are supported:
...
+     type: String
+     default: "500-599"
+     description: |
+        Comma-separated list of HTTP status codes that should be considered errors when returned by an HTTP client request.
```
Comment (mabdinur) [inline docs/configuration.rst:469]:
```suggestion
        Comma-separated list of HTTP status codes that should be considered errors when returned by an HTTP request.
```

File: tests/tracer/test_trace_utils.py [hunk 8]
```diff
@@ -505,6 +505,28 @@ def test_set_http_meta_custom_errors(mock_log, span, int_config, error_codes, st
         mock_log.exception.assert_not_called()
 
 
+@pytest.mark.subprocess(env={"DD_TRACE_HTTP_SERVER_ERROR_STATUSES": "404-412"})
+def test_set_http_meta_custom_errors_via_env():
+    from ddtrace import config
+    from ddtrace import tracer
+    from ddtrace.contrib.trace_utils import set_http_meta
+
+    config.http_server.error_statuses = "404-412"
```
Comment (mabdinur) [inline tests/tracer/test_trace_utils.py:514]:
```suggestion
```

**Review by rachelyangdog — APPROVED** [review]

File: docs/configuration.rst [hunk 9]
```diff
@@ -454,6 +454,13 @@ The following environment variables for the tracer are supported:
      default: True
      description: Send query strings in http.url tag in http server integrations.
 
+   DD_TRACE_HTTP_SERVER_ERROR_STATUSES:
+     type: String
+     default: "500-599"
+     description: |
+        Comma-separated list of HTTP status codes that should be considered errors when returned by an HTTP request.
+        The status codes are used to set the ``error`` field on the span.
```
Comment (erikayasuda) [inline docs/configuration.rst:462]: So comma separated and dashes? So they can mix like:
```
DD_TRACE_HTTP_SERVER_ERROR_STATUSES="200,400-404,500-599"
```
Maybe we should explicitly specify that they can also specify ranges with `-`?

**Review by erikayasuda — APPROVED** [review]

File: docs/configuration.rst (hunk 9 above)
Comment (mabdinur) [inline docs/configuration.rst:462]:
```suggestion
        The status codes are used to set the ``error`` field on the span. 
        Users can specific multiple comma separated error ranges (ex:  ``200,400-404,500-599``)
```

**Pull request merged**